src/components/dev-hub/academia-sign-up-form.js

```diff
@@ -66,6 +66,15 @@ const InstitutionSection = styled('div')`
     }
 `;
 
+const callRealm = async (data, callback) => {
+    try {
+        const res = await submitAcademiaForm(data);
+        res && callback(true);
+    } catch {
+        callback(false);
+    }
+};
```

Comment:
I like this idea because we abstract the `try/catch` to its own function but something feels a bit off to me in the `callback` piece taking true/false.

I think what I was expecting was something more like the Javascript Promise API, where there are separate handlers for `onSuccess` and `onFailure`. Then we also can case on when, for example, to run `setCanSubmit` a bit better. I think, for example, imagine we did the following:

```
const callRealm = async (data, onSuccess, onFailure) => {
    try {
        const res = await submitAcademiaForm(data);
        res && onSuccess();
    } catch {
        onFailure();
    }
};
```

then later (with the suggestion to avoid code on an unmounted component), we can simply do this:

```
callRealm(data,() => setSuccess(true), () => {setSuccess(false); setCanSubmit(true)};
```

Why I feel this is a bit better is:
- We clearly separate what code runs on a successful submission vs a failure
- We then can pass more to the `onFailure`, such as a failure message without having to worry about how general `callback` would be

Honestly though I'm not too opinionated on this, but I did get a weird vibe when looking through this and I think the callback use is why (or could we update `submitAcademiaForm` directly to add the try/catch maybe and remove this logic from here?) Idk there are a couple options

Reply:
i do like the idea of separating the two handlers better than using an if statement in a general `callback`! i think it also makes the code easier to understand. and yeah, i dont see the point in keeping the logic there, i can update `submitAcademiaForm` to have the try/catch